Sources/Request/Request/RequestParams/Query.swift

```diff
@@ -0,0 +1,46 @@
+//
+//  File.swift
+//  
+//
+//  Created by brennobemoura on 15/11/20.
+//
+
+import Foundation
+
+/// Sets the query string of the `Request`
+///
+/// `[key:value, key2:value2]` becomes `?key=value&key2=value2`
+public struct Query: RequestParam {
+    private let children: [QueryParam]
+
+    /// Creates the `Query` from `[key:value]` pairs
+    /// - Parameter params: Key-value pairs describing the `Query`
+    public init(_ params: [String: Any?]) {
+        children = params.map {
+            QueryParam($0.key, value: $0.value)
+        }
+    }
+
+    /// Creates the `Query` directly from an array of `QueryParam`s
+    public init(@RequestBuilder _ params: () -> RequestParam) {
```

Comment:
This should have it's own builder, which should probably be done in a separate PR. Please leave the array initializer for compatibility.